kahuna/public/js/components/gr-chips/gr-filter-chip.html

```diff
@@ -0,0 +1,21 @@
+<gr-datalist gr-search="$grChipsCtrl.getSuggestions($chip)"
+             gr-on-select="$grFilterChipCtrl.apply($value)">
+    <button type="button"
+            class="gr-filter-chip__type"
+            ng:click="$grFilterChipCtrl.toggleFilterType()"
+    >{{$chip.filterType === 'inclusion' ? '+' : '−' }}</button
+    ><span class="gr-filter-chip__key">{{$chip.key}}:</span
+    ><input type="text"
+            class="gr-filter-chip__value"
+            gr-datalist-input
+            gr-auto-width
+            ng:trim="false"
```

Comment:
If the ngModel value is trimmed, it's impossible for `grAutoWidth` to calculate the width required for the field when you have leading or trailing spaces. There is a [comment to that effect](https://github.com/guardian/grid/pull/1780/files#diff-e04ed75395af4a024b163a60173e4d07R5) by that directive though arguably it's a little far from the code here, but I was trying to avoid polluting templates with such comments.

I tried to automatically set ngTrim in the directive but it doesn't seem to be possible :-(